cluster-autoscaler/cloudprovider/azure/azure_agent_pool.go

```diff
@@ -55,7 +55,7 @@ type AgentPool struct {
 func NewAgentPool(spec *dynamic.NodeGroupSpec, az *AzureManager) (*AgentPool, error) {
 	as := &AgentPool{
 		azureRef: azureRef{
-			Name: spec.Name,
+			Name: strings.ToLower(spec.Name),
```

Comment:
lowing doesn't work, because Azure cloud provider doesn't do this. Cluster autoscaler need to match values from Azure cloud provider.